### ISSUE TITLE
Fix no-disable regression for some rules

### ISSUE THREAD

**Issue by dargmuesli** [stylelint/stylelint#6015]
### What steps are needed to reproduce the bug?

Use e.g.
```
.ProseMirror { /* stylelint-disable-line selector-class-pattern */
  ...
}
```

(taken from https://github.com/maevsi/maevsi/blob/1d6fc6eff10c922b028b8811d0aad25cad6fe85c/nuxt/assets/css/main.css#L90)

### What Stylelint configuration is needed to reproduce the bug?

```
module.exports = {
  extends: [
    'stylelint-config-standard',
    'stylelint-config-prettier',
    'stylelint-config-recommended-vue',
  ],
  plugins: ['stylelint-no-unsupported-browser-features'],
  rules: {
    'at-rule-no-unknown': [
      true,
      {
        ignoreAtRules: [
          'apply',
          'layer',
          'responsive',
          'screen',
          'tailwind',
          'variants',
        ],
      },
    ],
    'declaration-block-trailing-semicolon': null,
    'no-descending-specificity': null,
    'plugin/no-unsupported-browser-features': [true, { severity: 'warning' }],
  },
}

```

### How did you run Stylelint?

`stylelint **/*.{vue,css} --ignore-path .gitignore`

### Which version of Stylelint are you using?

`14.7.0`

### What did you expect to happen?

The linter error is not supposed to show up.

### What actually happened?

I receive the error `Expected class selector to be kebab-case  selector-class-pattern` as if the ignore comment wasn't there. The ignore worked previously up until `14.6.1`.

See Renovate PR https://github.com/maevsi/maevsi/pull/583 with failing GitHub Action https://github.com/maevsi/maevsi/runs/6043511344?check_suite_focus=true#step:5:1294.

**Comment by ybiquitous** [comment]
@dargmuesli Thank you for opening the issue and using the template.

I can reproduce the behavior on the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AFAThAzjAsgSzQzQAJhiB6AKmMwBcBPAGxkbwDtaBaKPTAQwBGzTqzYwaLGGFoQ0nMIz6ZMnAA59atGGjbFK5ADq6K1JGb2G2AXyMgANCEz0OfAB4Jwy++AhsAZngA5h7ARsTEBiAwrlpsUJiR8BGODMyiXJD+QZx0fHF8aFCRRjZsIFZAA).

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by ybiquitous** [comment]
Note: The actual `selector-class-pattern` setting is in the `stylelint-config-standard` config:

```js
'selector-class-pattern': [
	'^([a-z][a-z0-9]*)(-[a-z0-9]+)*$',
	{
		message: 'Expected class selector to be kebab-case',
	},
],
```

https://github.com/stylelint/stylelint-config-standard/blob/089a5351a9ddf2396e2ea7b940c2368e9c60156e/index.js#L129-L134

**Comment by dargmuesli** [comment]
Hmm, I'm wondering why an update to stylelint itself causes this issue. The `stylelint-config-standard` config hasn't been updated for half a year.

**Comment by jeddy3** [comment]
This is an interesting regression.

It seems limited to the `selector-class-pattern` rule. For some reason, it's not picked up in scoped disable commands.

I did a little investigating and there's a missing gap [in the demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcID0AqABAZwC4E8A2MeAlgHZYC0URGAhgEYHkkwAeFxzmhMYWEATuTB4aGDOQAONLFhj8SaFEgA6JAHQAFfhAwwAskX7b+aYKrRpUaNTcUqFaSHgHw0AYgBmHgNyqAvqogADQgGDhkNCwI4GLB4BAkHkQA5tFmDsogrLIkUBiZrpnY+ISkFJCJKeTYNLk0-FCZ-oF+QA) results compared to other rules:

<img width="576" alt="Screenshot 2022-04-16 at 10 30 01" src="https://user-images.githubusercontent.com/808227/163669949-cd63cdff-fc05-4442-bb19-d8af37ff57b2.png">

And the rule name isn't included when hoving over the warning in the VS Code extension.

I can't think what may be causing it, though. Any ideas?

@dargmuesli As a workaround, you can use an unscoped disable command, i.e. `/* stylelint-disable-line */`, but this will turn all the rules off for the line.

**Comment by jeddy3** [comment]
Upon further digging, it looks like other rules are affected too. In [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcID0AqABAZwC4E8A2MeAlgHZYC0URGAhgEYHkkwAeFxzmhMYWEATuTB4aGDOQAONLFhj8SAGi4FeA8gFsaLISLFKM3VYKJRJ02fP2G+gzdpNoUSADokAdADEIENAGIAQjT8aMCuaGioaG5uji4kAL6uIAogGDhkWgjgYsngECQAZkQA5lmhJOHOIKyyJFAYVfBoVdj4hKQUkIUl5Ng0dUFQVQphzSD8AK4EDQghoy3WanY6ojNNAAwjFWMGKjYaWuQmjWjro4kJSfFAA), `selector-max-class`, `selector-class-pattern` and `selector-max-id` can't be disabled whereas `selector-id-pattern` can. I'm at a loss for the underlying cause.

**Comment by boris-petrov** [comment]
I'm hitting `selector-max-universal` and `media-feature-name-no-vendor-prefix` as well.

**Comment by jeddy3** [comment]
> I'm hitting `selector-max-universal` and `media-feature-name-no-vendor-prefix` as well.

It looks to be a regression in the broader disable ranges then. I've labelled the issue as help wanted in case anyone has time to investigate.
